### ISSUE TITLE
Profiling many spawn and sync does not work on Windows

### ISSUE THREAD

**Issue by tkf** [JuliaLang/julia#43124]
@DilumAluthge noticed that the test I added in #43072 fails sometimes (presumably only) on Windows https://github.com/JuliaLang/julia/pull/43072#issuecomment-972569162

The test is to run this script

https://github.com/JuliaLang/julia/blob/455236eb7f95901f1c68d3c1cd17928e6c7f6fce/test/profile_spawnmany_exec.jl#L5-L14

and check that it does not error or time out. There's no backtrace from this script (yet) because I forgot to redirect stdout and stderr.

**Comment by tkf** [comment]
> I get the log output
> 
> ```
> ┌ Error: Failed test: "spawn and wait *a lot* of tasks in @profile" (n = 2000000)
> │ 
> ...
> │   proc.exitcode = 3221225477
> │   proc.termsignal = 0
> 
> ```
> 
> in https://build.julialang.org/#/builders/65/builds/5543
> 
> 3221225477 is `STATUS_ACCESS_VIOLATION` so there's something wrong. But it looks like there's no backtrace.
> 
> Note that `n = 2000000` is the largest number of tasks in the test. I wonder if ignoring this case is sufficient.
>
> --- https://github.com/JuliaLang/julia/issues/43125#issuecomment-972703601

**Comment by tkf** [comment]
@vtjnash noticed that Windows CI can hit the timeout too (even after #43125):

```
Error in testset threads:
Test Failed at C:\buildbot\worker-tabularasa\tester_win64\build\share\julia\test\threads.jl:214
  Expression: !timeout
```

https://build.julialang.org/#/builders/65/builds/5821

**Comment by tkf** [comment]
Oops, my PR comment in #43247 contained a substring `fixes #43124`.

**Comment by LilithHafner** [comment]
This also fails somtimes on mac (see https://buildkite.com/julialang/julia-master/builds/24520#0188847a-6af8-4be7-92cd-c03932978df4/591-594)

```
┌ Error: A "spawn and wait lots of tasks" test failed
│   n = 2000000
│   proc.exitcode = 0
│   proc.termsignal = 15
│   success(proc) = false
│   timeout = true
└ @ Main.Test31Main_threads /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-4.0/build/default-macmini-x64-4-0/julialang/julia-master/julia-f407a4cac3/share/julia/test/threads.jl:313
  threads                                           (1) \|         failed at 2023-06-04T00:31:23.823
Test Failed at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-4.0/build/default-macmini-x64-4-0/julialang/julia-master/julia-f407a4cac3/share/julia/test/threads.jl:319
  Expression: success(proc)
   
Test Failed at /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-x64-4.0/build/default-macmini-x64-4-0/julialang/julia-master/julia-f407a4cac3/share/julia/test/threads.jl:320
  Expression: !timeout
```

on `test x86_64-apple-darwin`
